api/docs/tools.md

```diff
@@ -0,0 +1,331 @@
+# Tools
+
+## What are "tools"?
+
+The name is unfortunately generic but this document refers to the command line tools that reside in [workbench-api.yaml](https://github.com/all-of-us/workbench/blob/master/api/tools).
+They're usually used to invoke operational or one-off tasks that are more complex than a small bash script but 
+haven't made it into our standard code base yet. The tools are able to compile and use classes from the 
+standard code base so they are relatively quick to write but difficult to maintain. It's not uncommon for a tool to silently break because one of its dependencies in
+the standard code base has changed. Point being, it's great that we have the capability to write these tools but don't 
+rely on them too much for more complex tasks that we expect to invoke regularly.
+
+Some of the best use cases are operational tasks that would otherwise require manual SQL queries or tasks that 
+need to invoke an external API call.
+
+Some examples of tools we've used  
+- Changing billing permissions for all users
+- Setting a user's authority roles 
+  - `./project.rb set-authority`
+- Listing runtimes in an environment 
+  - `./project.rb list-runtimes`
+
+Advantages of writing tools
+- It provides a quick way to write a script that can leverage much of the standard code base
+- It handles environment setup, including authentication and database connections
+
+Disadvantages
+- No automated testing
+- No compilation checks
```

Comment:
I run `./gradlew compileGeneratedJava compileTestJava compile__tools__java` on a regular basis to check that these are in compliance